packages/core/src/serializer.ts

```diff
@@ -20,14 +21,58 @@ export interface PrimitiveTypeSerializer {
     serializeValue(value: unknown, property: Property): string | undefined
 }
 
+const isPrimitiveTypeSerializer = (value: unknown): value is PrimitiveTypeSerializer =>
+       typeof value === "object"
+    && value !== null
+    && "serializeValue" in value
+    && typeof value.serializeValue === "function"
+
+
+/**
+ * Type to provide options to the serializer.
+ */
+export type SerializationOptions = Partial<{
+    /**
+     * Determines whether empty feature values are explicitly serialized or skipped during serialization.
+     * (The specification states that empty feature values SHOULD be serialized, but not that they MUST be.)
+     * Default = true, meaning that empty feature values are *not* skipped.
+     */
+    serializeEmptyFeatures: boolean
+
+    primitiveTypeSerializer: PrimitiveTypeSerializer
+}>
+
+
+/**
+ * @return the {@link MetaPointer} for the given {@link Feature}.
+ */
+export const metaPointerFor = (feature: Feature): MetaPointer => {
+    const {language} = feature.classifier
+    return {
+        language: language.key,
+        version: language.version,
+        key: feature.key
+    }
+}
+
+
 /**
  * @return a {@link SerializationChunk} of the given model (i.e., an array of {@link Node nodes} - the first argument) to the LionWeb serialization JSON format.
  */
 export const serializeNodes = <NT extends Node>(
     nodes: NT[],
     extractionFacade: ExtractionFacade<NT>,
-    primitiveTypeSerializer: PrimitiveTypeSerializer = new DefaultPrimitiveTypeSerializer()
+    primitiveTypeSerializerOrOptions?: PrimitiveTypeSerializer | SerializationOptions
 ): SerializationChunk /* <=> JSON */ => {
+    const primitiveTypeSerializer = (
+        isPrimitiveTypeSerializer(primitiveTypeSerializerOrOptions)
+            ? primitiveTypeSerializerOrOptions
+            : primitiveTypeSerializerOrOptions?.primitiveTypeSerializer
+    ) ?? new DefaultPrimitiveTypeSerializer()
+    const serializeEmptyFeatures = isPrimitiveTypeSerializer(primitiveTypeSerializerOrOptions)
```

Comment:
here shouldn't we look at primitiveTypeSerializer and not primitiveTypeSerializerOrOptions?

Reply:
Line 68 checks whether `primitiveTypeSerializerOrOptions` *is* a `PrimitiveTypeSerializer`.